main.c

```diff
@@ -479,6 +498,158 @@ static struct kanshi_config *read_config(const char *config) {
 	return parse_config(config_path);
 }
 
+static void destroy_config(struct kanshi_config *config) {
+	struct kanshi_profile *profile, *tmp_profile;
+	wl_list_for_each_safe(profile, tmp_profile, &config->profiles, link) {
+		struct kanshi_profile_output *output, *tmp_output;
+		wl_list_for_each_safe(output, tmp_output, &profile->outputs, link) {
+			free(output->name);
+			wl_list_remove(&output->link);
+			free(output);
+		}
+		struct kanshi_profile_command *command, *tmp_command;
+		wl_list_for_each_safe(command, tmp_command, &profile->commands, link) {
+			free(command->command);
+			wl_list_remove(&command->link);
+			free(command);
+		}
+		wl_list_remove(&profile->link);
+		free(profile);
+	}
+	free(config);
+}
+
+static bool reload_config(struct kanshi_state *state) {
+	fprintf(stderr, "reloading config\n");
+	struct kanshi_config *config = read_config();
+	if (config != NULL) {
+		destroy_config(state->config);
+		state->config = config;
+		state->pending_profile = NULL;
+		state->current_profile = NULL;
+		return try_apply_profiles(state);
+	}
+	return false;
+}
+
+static int do_poll(struct pollfd *fds, nfds_t nfds) {
+	int ret;
+	do {
+		ret = poll(fds, nfds, -1);
+	} while (ret == -1 && errno == EINTR);
+	return ret;
+}
+
+enum readfds_type {
+	FD_WAYLAND		= 0,
+#ifdef KANSHI_HAS_VARLINK
+	FD_VARLINK		= 1,
+#endif
+	FD_COUNT
+};
+
+static int loop(struct kanshi_state *state) {
```

Comment:
I'd prefer to return `bool` values to indicate success/failure

Reply:
This makes the logic slightly more complicated in `main`, which does `ret = kanshi_main_loop`. Not sure it'd be worth it?

Reply:
Fair enough